assets/js/googlesitekit/data/create-snapshot-store.js

```diff
@@ -38,12 +39,15 @@ const SET_STATE_FROM_SNAPSHOT = 'SET_STATE_FROM_SNAPSHOT';
  * Creates a store object that includes actions and controls for restoring/creating state snapshots.
  *
  * @since 1.9.0
+ * @since n.e.x.t Added the ability to pick specific parts of the state to save in the snapshot.
  * @private
  *
- * @param {string} storeName The name of the store to snapshot in the cache.
+ * @param {string}          storeName      The name of the store to snapshot in the cache.
+ * @param {Object}          [options]      Optional configuration object.
+ * @param {string|string[]} [options.pick] Property path(s) to pick from state when creating snapshots.
  * @return {Object} The snapshot store object.
  */
-export function createSnapshotStore( storeName ) {
+export function createSnapshotStore( storeName, { pick: pickPaths } = {} ) {
```

Comment:
Rather than using `pick`/`pickPaths`, can we have a more descriptive name for this argument? Calling it `pick` seems like the implementation detail of using `lodash`'s `pick()` is bubbling up.

I'd call this `keysToSnapshot` or `onlySnapshotKeys`, etc. so it's a bit more clear to someone reading it.